kyo-core/jvm/src/main/scala/kyo/Path.scala

```diff
@@ -339,22 +354,55 @@ object Path:
 
     type Part = String | Path
 
-    def apply(path: List[Part]): Path =
-        def loop(path: List[Part], acc: List[String]): List[String] =
-            path match
-                case _: Nil.type =>
-                    acc.reverse
-                case h :: t =>
-                    h match
-                        case h: String =>
-                            loop(t, h.split('/').filter(_.nonEmpty).toList.reverse ::: acc)
-                        case h: Path =>
-                            loop(h.path ::: t, acc)
-        new Path(loop(path, Nil))
+    def apply(parts: List[Part]): Path =
+        val flattened = parts.flatMap {
+            case p if isNull(p) => Nil
+            case s: String      => List(s)
+            case p: Path        => p.path
+        }
+        val javaPath       = if flattened.isEmpty then Paths.get("") else Paths.get(flattened.head, flattened.tail*)
+        val normalizedPath = javaPath.normalize().toString
+
+        new Path(if normalizedPath.isEmpty then Nil else normalizedPath.split("/").toList)
     end apply
 
     def apply(path: Part*): Path =
         apply(path.toList)
+
+    object Base:
+        def cache(using Frame): Path < IO      = IO(Path(BaseDirectories.get().cacheDir))
+        def config(using Frame): Path < IO     = IO(Path(BaseDirectories.get().configDir))
+        def data(using Frame): Path < IO       = IO(Path(BaseDirectories.get().dataDir))
+        def dataLocal(using Frame): Path < IO  = IO(Path(BaseDirectories.get().dataLocalDir))
+        def executable(using Frame): Path < IO = IO(Path(BaseDirectories.get().executableDir))
+        def preference(using Frame): Path < IO = IO(Path(BaseDirectories.get().preferenceDir))
+        def runtime(using Frame): Path < IO    = IO(Path(BaseDirectories.get().runtimeDir))
+    end Base
```

Comment:
Wouldn't it be better to create a Kyo's own `case class BaseDirectories(cache: Path, config: Path, ...)`.
And on `BaseDirectories` companion object have method
```scala
lazy val get: BaseDirectories < IO = ...
```

And the same for the other 2 cases.

Reply:
Makes sense, I've updated it. I can't be a `lazy val` because of `Frame`, though.

Reply:
Great, now it's perfect, thanks :pray: